src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1305,13 +1307,204 @@ public static Matrix4x4 CreateReflection(Plane value)
                    d * (e * jo_kn - f * io_km + g * in_jm);
         }
 
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<float> Permute(Vector128<float> value, byte control)
+        {
+            if (Avx.IsSupported)
+            {
+                return Avx.Permute(value, control);
+            }
+
+            Debug.Assert(Sse.IsSupported);
+            return Sse.Shuffle(value, value, control);
+        }
+
         /// <summary>
         /// Attempts to calculate the inverse of the given matrix. If successful, result will contain the inverted matrix.
         /// </summary>
         /// <param name="matrix">The source matrix to invert.</param>
         /// <param name="result">If successful, contains the inverted matrix.</param>
         /// <returns>True if the source matrix could be inverted; False otherwise.</returns>
-        public static bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
+        ///
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
+        {
+            if (Sse.IsSupported)
+            {
+                return SseImpl(matrix, out result);
+            }
+
+            return SoftwareFallback(matrix, out result);
+        }
+
+
+
+        private static unsafe bool SseImpl(Matrix4x4 matrix, out Matrix4x4 result)
```

Comment:
This and the other should still be made `static local functions` as I called out here: https://github.com/dotnet/runtime/pull/36323#discussion_r426401821

Otherwise they may conflict with other methods we also accelerate.

Reply:
Everything thing else LGTM and I believe it is otherwise ready to merge

Reply:
Sorry, wasn't familiar with local functions.  I thought you were pointing out the public modifier instead.  Fixed!

Reply:
No worries, I should have clarified 😄